Revise configuration construction

nrempel opened the pull request:
Closes #1 

This change removes `confy` completely. I gave some rationale for that in #1. This also removes the need for a default implementation for `EppClientConfig ` this seemed awkward to me since it should be impossible to create a client without a config.

Instead of using `lazy_static`, the config is instantiated and passed from the caller.

Finally, I removed the use of `mpsc` in `connect` since I think the didn't provide much benefit since connect is only called once per client.

nrempel commented:
Just letting you know I plan on rebasing and picking this back up once #3 is merged.

masalachai commented:
> Just letting you know I plan on rebasing and picking this back up once #3 is merged.

Thanks. I have merged the PR.

nrempel commented:
I think this should be ready. I updated the docs and README as well. I also cleaned up the doctests and added the doc tests to the CI build. They don't run but they should check to make sure they compile.

masalachai commented:
Thanks! I will also bump the version no after merging this and then probably start on the change to the public API mentioned in #4

nrempel commented:
> Thanks! I will also bump the version no after merging this and then probably start on the change to the public API mentioned in #4

Sounds good!
